Apps/Laboratory/src/Services/LaboratoryService.cs

```diff
@@ -86,224 +86,163 @@ public class LaboratoryService : ILaboratoryService
-                return retVal;
+                return RequestResultFactory.Error<PublicCovidTestResponse>(UnauthorizedResultError());
             }
 
-            RequestResult<PhsaResult<IEnumerable<CovidTestResult>>> result = await this.laboratoryDelegate.GetPublicTestResults(accessToken, phn, dateOfBirth, collectionDate).ConfigureAwait(true);
-            List<CovidTestResult> payload = result.ResourcePayload?.Result?.ToList() ?? new List<CovidTestResult>();
-            PhsaLoadState? loadState = result.ResourcePayload?.LoadState;
+            RequestResult<PhsaResult<IEnumerable<CovidTestResult>>> result =
+                await this.laboratoryDelegate.GetPublicTestResults(accessToken, phn, DateOnly.FromDateTime(dateOfBirth), DateOnly.FromDateTime(collectionDate)).ConfigureAwait(true);
 
-            retVal.ResultStatus = result.ResultStatus;
-            retVal.ResultError = result.ResultError;
-
-            if (payload.Any())
+            PhsaLoadState? loadState = result.ResourcePayload?.LoadState;
+            if (loadState != null && loadState.RefreshInProgress)
             {
-                LabIndicatorType labIndicatorType = Enum.Parse<LabIndicatorType>(payload.Select(x => x.StatusIndicator).First());
-
-                switch (labIndicatorType)
-                {
-                    case LabIndicatorType.Found:
-                        retVal.ResourcePayload = new PublicCovidTestResponse
-                        {
-                            Records = this.autoMapper.Map<List<CovidTestResult>, IEnumerable<PublicCovidTestRecord>>(payload),
-                        };
-                        break;
-
-                    case LabIndicatorType.DataMismatch:
-                    case LabIndicatorType.NotFound:
-                        retVal.ResultStatus = ResultType.ActionRequired;
-                        retVal.ResultError = ErrorTranslator.ActionRequired(ErrorMessages.DataMismatch, ActionType.DataMismatch);
-                        break;
-
-                    case LabIndicatorType.Threshold:
-                    case LabIndicatorType.Blocked:
-                        retVal.ResultStatus = ResultType.ActionRequired;
-                        retVal.ResultError = ErrorTranslator.ActionRequired(ErrorMessages.RecordsNotAvailable, ActionType.Invalid);
-                        break;
-                }
+                return RequestResultFactory.ActionRequired(
+                    new PublicCovidTestResponse
+                    {
+                        RetryIn = Math.Max(loadState.BackOffMilliseconds, this.labConfig.BackOffMilliseconds),
+                        Loaded = !loadState.RefreshInProgress,
+                    },
+                    ActionType.Refresh,
+                    "Refresh in progress");
             }
 
-            if (loadState != null)
+            LabIndicatorType labIndicatorType = Enum.Parse<LabIndicatorType>(result.ResourcePayload?.Result.FirstOrDefault()?.StatusIndicator);
```

Comment:
`Enum.Parse` will throw an exception if it's passed a null value, which will happen if there are no results.

Reply:
So would the original -  Enum.Parse<LabIndicatorType>(payload.Select(x => x.StatusIndicator).First());
I'll make it null safe, though we need to revisit our null payload approach.